packages/valory/skills/price_estimation_abci/rounds.py

```diff
@@ -144,6 +154,61 @@ class TxHashRound(CollectSameUntilThresholdRound):
     selection_key = get_name(SynchronizedData.most_voted_tx_hash)
 
 
+class DataHashSignRound(CollectDifferentUntilAllRound):
+    """A round in which agents sign the data hash"""
+
+    payload_class = SignaturePayload
+    payload_attribute = "signature"
+    synchronized_data_class = SynchronizedData
+    done_event = Event.DONE
+    none_event = Event.NONE
+    no_majority_event = Event.NO_MAJORITY
+    collection_key = get_name(SynchronizedData.service_data_signatures)
+
+    def check_payload(self, payload: BaseTxPayload) -> None:
+        """Check Payload"""
+        if payload.round_count != self.synchronized_data.round_count:
+            raise TransactionNotValidError(
+                f"Expected round count {self.synchronized_data.round_count} and got {payload.round_count}."
+            )
+
+        if payload.sender in self.collection:
+            raise TransactionNotValidError(
+                f"sender {payload.sender} has already sent value for round: {self.round_id}"
+            )
```

Comment:
Why is this necessary?

Reply:
@0xArdi sorry used cause test was incorrect.

Reply:
done, thanks